### PR TITLE
Issue/12251 add site title qs task

### LINKED ISSUE #12251: [Quick Start] Add initial "Set a site title" step
Since we removed choosing a site title from the site creation flow, we'd like to move it to be the first step in the Quick Start process.

**Gridicon:** pencil
**Title:** "Set your site title"
**Description:** "Give your site a name that reflects its personality and topic. First impressions count!"

**Steps**

* "Tap _Site Title_ to set a new title" (replace Site Title with the current site name, and show a spotlight on it in the site header)
* The usual popup menu will be displayed, and a spotlight will be shown on the site title row in Site Settings, as per #12250. Visiting the Site title screen will check off this tour.

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #12251

This PR depends on the [following](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1622) Flux-C PR. It needs to be merged first, then Flux-C reference updated.

In this PR we added a new Quick Start action for updating a Site Title from My Site screen.

[Image from Gyazo](https://gyazo.com/4e1cdfa6eec0accffc168af049be6514)

[Image from Gyazo](https://gyazo.com/eca1b6843e7291a7420cf852a172e4b1)

To test:
- Create a new site and start Quick Start.
- Navigate to `Customize Your Site` QS category and select `Set your site title` task.
- Notice that the QS Snackbar mentions site title, and the focus point is displayed at site title.
- Tap on the site title, and make sure the task is marked as completed. You don't need to change the actual title to complete the task.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
Hi there! `15.2` is being frozen, so this PR is being bumped to `15.3`. If these changes _have to_ be shipped with `15.2`, please target `release/15.2` and cc me in the PR so I can cut a new beta release. Also if you can keep the release dates in mind and bump the PRs & issues you are working on before the code freeze date, we'd really appreciate it.

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt:1380]: NIT: I'm wondering if it might be a little cleaner to build the message separately to remove some of the redundant code, something like: 
```
val actionName = if (activeTutorialPrompt?.task == UPDATE_SITE_TITLE) {
            my_site_title_label.text.toString()
        } else null
        val shortQuickStartMessage =
                    stylizeQuickStartPrompt(
                            requireActivity(),
                            activeTutorialPrompt!!.shortMessagePrompt,
                            activeTutorialPrompt!!.iconId,
                            actionName
                    )
```
However, it's not a big deal so feel free to use what you feel is better :-)

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartMySitePrompts.kt
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartMySitePrompts.kt:21]: How would you feel about extracting the `-1` into a constant? Something like `ICON_NOT_SET`

File: WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt:69]: Maybe also use the `ICON_NOT_SET` here?

**Review by planarvoid — CHANGES_REQUESTED** [review]
Thanks for this PR! It looks mostly great. I've found one bug though. When you update the title, the QuickStart block doesn't get automatically updated (there is still "1 of 7 complete"). I think we need to update the block on site title update. Does it make sense? The rest of my comments is minor :-)
quick_start_not_updated

**Review by osullivanchris — CHANGES_REQUESTED** [review]
Hey mostly looks good! Unlike @planarvoid mine does say 2/7 complete. However, it didn't go on to the next step automatically like the other ones do. It should present me with the next step automatically after this one is complete.

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt:1381]: Great idea! I implemented it in a bit different way - since `UPDATE_SITE_TITLE` message is one-off, I decided to style it separately, in order to not overcomplicate the logic in `stylizeQuickStartPrompt`.

File: WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartMySitePrompts.kt
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/quickstart/QuickStartMySitePrompts.kt]: Sounds good to me 👍

**Comment by khaykov** [comment]
@planarvoid @osullivanchris Thank you so much for the review! I worked on QS a long time ago and forgot that the dialogs need a special logic that triggers the next step and updates the number of completed tasks 🤦 Should be fixed now!

**Comment by planarvoid** [comment]
thanks for the changes @khaykov ! Now it works well 👍

**Review by planarvoid — APPROVED** [review]
thanks for the changes @khaykov ! Now it works well 👍 . Feel free to merge it @osullivanchris once you're happy with the behaviour :-)

**Comment by osullivanchris** [comment]
Yeah that's all I needed! Thanks for the fix @khaykov !

**Comment by oguzkocer** [comment]
@osullivanchris You'll need to approve the PR for the merge button to get enabled. Today is code-freeze day, so if you'd like this PR to make it to the code freeze, it needs to be merged in the next hour or so. cc @planarvoid @khaykov

**Review by osullivanchris — APPROVED** [review]

**Comment by khaykov** [comment]
I got ok from @oguzkocer and @planarvoid to go ahead with and merge this one 🚀

**Pull request merged**
